modelkit/assets/drivers/azure.py

```diff
@@ -0,0 +1,87 @@
+import os
+from typing import Optional
+
+from azure.storage.blob import BlobServiceClient
+from structlog import get_logger
+from tenacity import retry
+
+from modelkit.assets import errors
+from modelkit.assets.drivers.abc import StorageDriver
+from modelkit.assets.drivers.retry import RETRY_POLICY
+
+logger = get_logger(__name__)
+
+
+class AzureStorageDriver(StorageDriver):
+    bucket: str
+
+    def __init__(
+        self,
+        bucket: Optional[str] = None,
+        connection_string: Optional[str] = None,
+        client: Optional[BlobServiceClient] = None,
+    ):
+        self.bucket = bucket or os.environ.get("MODELKIT_STORAGE_BUCKET") or ""
```

Comment:
NB: the final `or ""` is not needed

Reply:
It in fact is important, because `bucket` is of type `str` and somehow  `mypy` complains with `os.environ.get("...", "")`

Reply:
* yells at cloud *